lib/commands/deprecate.js

```diff
@@ -0,0 +1,64 @@
+'use strict'
+
+const Deprecate = module.exports = {
+  command: 'deprecate <pkg>[@<version>] <message>',
+  describe: 'Deprecate a version of a package',
+  builder (y) {
+    return y.help().alias('help', 'h')
+      .options(Deprecate.options)
+  },
+  options: Object.assign(require('../common-opts.js', {})),
+  handler: async argv => deprecate(argv)
+}
+
+async function deprecate (argv) {
+  const figgyPudding = require('figgy-pudding')
+  const BB = require('bluebird')
+  const libnpm = require('libnpm')
+  const npmConfig = require('../config.js')
+  const semver = require('semver')
+  // const otplease = require('./utils/otplease.js')
+
+  const DeprecateConfig = figgyPudding({
+    json: {},
+    loglevel: {},
+    parseable: {},
+    silent: {}
+  })
+
+  const opts = DeprecateConfig(npmConfig().concat(argv).concat({
+    log: require('npmlog')
+  }))
+
+  return BB.try(() => {
+    // fetch the data and make sure it exists.
+    const p = libnpm.parseArg(argv['pkg@version'])
```

Comment:
This is so strange. Is this seriously handled just like that by yargs?!

Reply:
I first thought I would have to manually join the package name and the version, but then I ran the debugger and spotted that:
![annotation 2018-11-09 170846](https://user-images.githubusercontent.com/5957867/48290851-664b8580-e442-11e8-96b6-96f2f0731280.png)

Fortunately it seems to work the way we want, but even the [yargs docs](https://github.com/yargs/yargs/blob/master/docs/advanced.md#positional-arguments) don't seem to document what happens when two positional arguments are character-separated instead of whitespace-separated. I can't find a more comprehensive document for how it parses command templates.

Reply:
From what I can tell from [the code](https://github.com/yargs/yargs/blob/331cb203ecc95348e2b547d51ecece6f2e7271d2/lib/command.js) it seems to just consider those cases as one positional argument, which makes sense.